compute_tools/src/pg_helpers.rs

```diff
@@ -248,18 +248,20 @@ pub fn wait_for_postgres(pg: &mut Child, port: &str, pgdata: &Path) -> Result<()
             bail!("Postgres exited unexpectedly with code {}", code);
         }
 
-        if pid_path.exists() {
-            let file = BufReader::new(File::open(&pid_path)?);
-            let status = file
-                .lines()
-                .last()
-                .unwrap()
-                .unwrap_or_else(|_| "unknown".to_string());
-            let can_connect = TcpStream::connect_timeout(&addr, timeout).is_ok();
-
-            // Now Postgres is ready to accept connections
-            if status.trim() == "ready" && can_connect {
-                break;
+        // Check that we can open pid file first.
+        if let Ok(file) = File::open(&pid_path) {
+            let file = BufReader::new(file);
+            let last_line = file.lines().last();
+
+            // Pid file could be there and we could read it, but it could be empty, for example.
+            if let Some(line) = last_line {
+                let status = line.unwrap_or_else(|_| "unknown".to_string());
```

Comment:
NIT (not your PR caused this anyway): `to_string` will cause an allocation, and `trim` is actually available on `&str`, so we could do below smth. like
`if line.map(|line| line.trim()) == Some("ready")`.

Reply:
My bad, I've thought it's a pure String in the `Some(line)` code hence all the suggestions.
We can use `as_deref()` on Option/Result to override this:

```rust
// Pid file could be there and we could read it, but it could be empty, for example.
if let Some(line) = last_line {
    let status = line.as_deref().unwrap_or("unknown").trim();
    let can_connect = TcpStream::connect_timeout(&addr, timeout).is_ok();

    // Now Postgres is ready to accept connections
    if status == "ready" && can_connect {
        break;
    }
}
```

Either way, not important, so feel free to ignore if it takes too much time.

Reply:
None, I was slightly suspicious that the semantics changes a bit: now, we throw away file reading errors, while before we substituted it with `"unknown"`, but IMO there's not much of a difference after I've given it a thought.